src/art.rs

```diff
@@ -1211,10 +1424,22 @@ impl<P: KeyTrait, V: Clone> Tree<P, V> {
                     return (Some(Arc::new(new_node)), true);
                 }
                 (None, true) => {
-                    // The key has been found, so remove the corresponding
-                    // child link from the current node.
-                    let new_node = cur_node.delete_child(k);
-                    return (Some(Arc::new(new_node)), true);
+                    // The key has been found, so either remove the corresponding
```

Comment:
Is it possible to explain this in detail with an example? I am just stuck a bit here